### PR TITLE
Improve code coverage of the polygon class by adding more tests to the test_polygon.py file

### PR DESCRIPTION
**Pull request by johan1505**
#18648 ## Brief description of what is fixed or changed

Improved the code coverage accomplished by the tests in the test_polygon.py file. Code coverage improved from 69% to 80%. To do this, I added two new tests called test_type_of_triangle and test_do_poly_distance. These tests reach statements in the Polygon class that were not being reached or executed by the existing tests. Also, I added some extra assertions/checks to the test_transform and test_cut_section tests.

#### Other comments
This pull request is the result of the works for a university project on Software Testing and Verification

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/geometry/tests/test_polygon.py [hunk 1]
```diff
@@ -611,3 +613,44 @@ def test_cut_section():
...
+    # Non-intersecting polygons
+    square1 = Polygon (Point(0,0), Point(0,1), Point(1,1), Point(1,0))
+    triangle1 = Polygon(Point(1, 2), Point(2, 2), Point(2, 1))
+    assert square1._do_poly_distance(triangle1) == sqrt(2)/2
+
+    # Polygons which sides intersect
+    square2 = Polygon(Point(1,0), Point(2,0), Point(2,1), Point(1,1))
+    with warns(UserWarning, \
+               match="Polygons may intersect producing erroneous output"):
+        assert square1._do_poly_distance(square2) == 0
```
Comment (sylee957) [inline sympy/geometry/tests/test_polygon.py]:
```suggestion
        assert square1._do_poly_distance(square2) == 0
```

File: sympy/geometry/tests/test_polygon.py [hunk 2]
```diff
@@ -611,3 +613,44 @@ def test_cut_section():
...
+    with warns(UserWarning, \
+               match="Polygons may intersect producing erroneous output"):
+        assert square1._do_poly_distance(square2) == 0    
+
+    # Polygons which bodies intersect
+    triangle2 = Polygon(Point(0, -1), Point(2, -1), Point(S.Half, S.Half))
+    with warns(UserWarning, \
+               match="Polygons may intersect producing erroneous output"):
+        assert triangle2._do_poly_distance(square1) == 0
+
```
Comment (sylee957) [inline sympy/geometry/tests/test_polygon.py]: You should not have an extra spacing in EOF

File: sympy/geometry/tests/test_polygon.py (hunk 1 above)
Comment (sylee957) [reply sympy/geometry/tests/test_polygon.py]: I don't think that this had been resolved yet

File: sympy/geometry/tests/test_polygon.py [hunk 3]
```diff
@@ -611,3 +613,43 @@ def test_cut_section():
...
     assert p3 == Polygon(Point2D(0, 10), Point2D(0, 0), Point2D(20, 0), Point2D(20, 10))
     assert p4 == None
+
+    # case where the line does not intersect with a polygon at all
+    raises(ValueError, lambda: p.cut_section(Line((0, a), slope=0)))
+
+def test_type_of_triangle():
+    # Isoceles triangle
+    p1 = Polygon(Point(0,0), Point(5,0), Point(2,4))
+    assert p1.is_isosceles() == True
```
Comment (Smit-create) [inline sympy/geometry/tests/test_polygon.py]: Please have a look at the PEP8 style and add spacing aroung the commas. For instance this should be `p1 = Polygon(Point(0, 0), Point(5,0), Point(2,4))`

File: sympy/geometry/tests/test_polygon.py [hunk 4]
```diff
@@ -611,3 +613,43 @@ def test_cut_section():
...
+    assert p1.is_scalene() == False
+    assert p1.is_equilateral() == False
+
+    # Scalene triangle
+    p2 = Polygon (Point(0,0), Point(0,2), Point(4,0))
+    assert p2.is_isosceles() == False
```
Comment (Smit-create) [inline sympy/geometry/tests/test_polygon.py]: +1

File: sympy/geometry/tests/test_polygon.py [hunk 5]
```diff
@@ -611,3 +613,43 @@ def test_cut_section():
...
+    assert p2.is_scalene() == True
+    assert p2.is_equilateral() == False
+
+    # Equilateral triagle
+    p3 = Polygon(Point(0,0), Point(6,0), Point(3, sqrt(27)))
+    assert p3.is_isosceles() == True
```
Comment (Smit-create) [inline sympy/geometry/tests/test_polygon.py]: +1, Apply PEP8 wherever it is applicable in the following lines of code

File: sympy/geometry/tests/test_polygon.py (hunk 1 above)
Comment (Smit-create) [reply sympy/geometry/tests/test_polygon.py]: @johan1505 Please remove the white spacing at the end of line 649, as this is why it fails the ci tests.

**Pull request merged**